### PR TITLE
DEVX-2074: Add source code links to the intros of affected client examples

### PR DESCRIPTION
**Pull request by tikimims**
Update affected client example intros with source code links

### CODE REVIEW

**Comment by ybyzek** [comment]
@tikimims changes look good.  Did you validate a docs build passes with this PR?  (Or would you like assistance to set that up?)

**Comment by tikimims** [comment]
@ybyzek Great! I'm working on validating the docs build. Are there certain changes I need to make outside of branching off from 6.0.x?

**Comment by tikimims** [comment]
@ybyzek Build is clean. This one is ready for your review.

File: clients/docs/kafka-connect-datagen.rst [hunk 1]
```diff
@@ -3,8 +3,10 @@
 Kafka Connect Datagen
 =====================
 
-In this tutorial, you will run a |kconnect-long| Datagen source connector that
-produces messages to and consumes messages from an |ak-tm| cluster.
+In this tutorial, you will run a |kconnect-long| Datagen source connector using
+`Kafka Connect Datagen
+<https://www.confluent.io/hub/confluentinc/kafka-connect-datagen?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud>`__
```
Comment (ybyzek) [inline clients/docs/kafka-connect-datagen.rst]:
```suggestion
<https://www.confluent.io/hub/confluentinc/kafka-connect-datagen>`__
```

**Review by ybyzek — APPROVED** [review]
Thanks @tikimims , I left one minor comment to be addressed before merge, and then the PR is ready to go.

**Comment by tikimims** [comment]
@ybyzek Thank you!! I committed the change and did a full build - clean. I think we're good to merge.

**Pull request merged**